contracts/YearnCrvTricrypto.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.8.5;
+
+import "./interfaces/ICrvTricrypto.sol";
+import "./interfaces/IERC20.sol";
+import "./interfaces/ITrigger.sol";
+import "./interfaces/IYVaultV2.sol";
+
+/**
+ * @notice Defines a trigger that is toggled if any of the following conditions occur:
+ *   1. The price per share for the V2 yVault significantly decreases between consecutive checks. Under normal
+ *      operation, this value should only increase. A decrease indicates something is wrong with the Yearn vault
+ *   2. Curve Tricrypto token balances are significantly lower than what the pool expects them to be
+ *   3. Curve Tricrypto virtual price drops significantly
+ */
+contract YearnCrvTricrypto is ITrigger {
+  // --- Tokens ---
+  // Token addresses
+  IERC20 internal constant usdt = IERC20(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+  IERC20 internal constant wbtc = IERC20(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+  IERC20 internal constant weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+  // Token indices in Curve pool arrays
+  uint256 internal constant usdtIndex = 0;
+  uint256 internal constant wbtcIndex = 1;
+  uint256 internal constant wethIndex = 2;
+
+  // --- Tolerances ---
+  /// @dev Scale used to define percentages. Percentages are defined as tolerance / scale
+  uint256 public constant scale = 1000;
```

Comment:
Just curious how you chose 10e3 as the scale factor for defining these percentages? Just kind of a "good enough" or was there a calculation involved to ensure rounding errors would necessarily fall below the threshold concerned at that scale?

Reply:
It's pretty arbitrary. The other Yearn trigger we used 1e18, but multiplying by e.g. 5e17 to divide by 1e18 felt unnecessary and more likely to result in overflow. So for this one I used 1000 to reduce overflow but still give enough precision to check e.g. a 50.1% decrease if you wanted